disperser/batcher/batch_confirmer.go

```diff
@@ -0,0 +1,536 @@
+	err := b.ChainState.Start(ctx)
+	if err != nil {
+		return err
+	}
+	// Wait for few seconds for indexer to index blockchain
+	// This won't be needed when we switch to using Graph node
+	time.Sleep(indexerWarmupDelay)
+
+	go func() {
+		receiptChan := b.TransactionManager.ReceiptChan()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case receiptOrErr := <-receiptChan:
+				b.logger.Info("received response from transaction manager", "receipt", receiptOrErr.Receipt, "err", receiptOrErr.Err)
+				err := b.ProcessConfirmedBatch(ctx, receiptOrErr)
+				if err != nil {
+					b.logger.Error("failed to process confirmed batch", "err", err)
+				}
+			}
+		}
+	}()
+	b.TransactionManager.Start(ctx)
+
+	go func() {
+		ticker := time.NewTicker(b.PullInterval)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				if err := b.HandleSingleBatch(ctx); err != nil {
+					b.logger.Error("failed to process a batch", "err", err)
+				}
+			}
+		}
+	}()
+
+	return nil
+}
+
+// updateConfirmationInfo updates the confirmation info for each blob in the batch and returns failed blobs to retry.
+func (b *BatchConfirmer) updateConfirmationInfo(
+	ctx context.Context,
+	batchData confirmationMetadata,
+	txnReceipt *types.Receipt,
+) ([]*disperser.BlobMetadata, error) {
+	if txnReceipt.BlockNumber == nil {
+		return nil, errors.New("error getting transaction receipt block number")
+	}
+	if len(batchData.blobs) == 0 {
+		return nil, errors.New("failed to process confirmed batch: no blobs from transaction manager metadata")
+	}
+	if batchData.batchHeader == nil {
+		return nil, errors.New("failed to process confirmed batch: batch header from transaction manager metadata is nil")
+	}
+	if len(batchData.blobHeaders) == 0 {
+		return nil, errors.New("failed to process confirmed batch: no blob headers from transaction manager metadata")
+	}
+	if batchData.merkleTree == nil {
+		return nil, errors.New("failed to process confirmed batch: merkle tree from transaction manager metadata is nil")
+	}
+	if batchData.aggSig == nil {
+		return nil, errors.New("failed to process confirmed batch: aggSig from transaction manager metadata is nil")
+	}
+	headerHash, err := batchData.batchHeader.GetBatchHeaderHash()
+	if err != nil {
+		return nil, fmt.Errorf("error getting batch header hash: %w", err)
+	}
+	batchID, err := b.getBatchID(ctx, txnReceipt)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching batch ID: %w", err)
+	}
+
+	blobsToRetry := make([]*disperser.BlobMetadata, 0)
+	var updateConfirmationInfoErr error
+
+	for blobIndex, metadata := range batchData.blobs {
+		// Mark the blob failed if it didn't get enough signatures.
+		status := disperser.InsufficientSignatures
+
+		var proof []byte
+		if isBlobAttested(batchData.aggSig.QuorumResults, batchData.blobHeaders[blobIndex]) {
+			status = disperser.Confirmed
+			// generate inclusion proof
+			if blobIndex >= len(batchData.blobHeaders) {
```

Comment:
It's unclear to me when this would ever happen. Isn't in at invariant that the lengths should be the same?

Reply:
Yeah, not sure why this was added before. 
If this condition were to ever happen, `batchData.blobHeaders[blobIndex]` would have resulted in out of bounds exception in L170. 
I don't see any errors logs from this block as well.
I'll remove this block.